vars/pod.groovy

```diff
@@ -31,16 +33,29 @@ def volumes() {
   ]
 }
 
-def slaveTemplate(name, image, shell) {
-    containerTemplate(
-        name: name,
-        image: image,
-        command: "$shell -c",
-        args: 'cat',
-        ttyEnabled: true,
-        workingDir: '/home/jenkins/',
-        resourceLimitMemory: '640Mi'
-    )
+def slaveTemplate(name, image, shell, envVars) {
+    if (envVars) {
+      return containerTemplate(
```

Comment:
What happens if we pass empty `envVars`:  like [ ] or `null`? 

If there is no issue then we can just pass an empty `envVars`, we can get rid of this if block.

Reply:
when we pass envVars: [ ] to conatinerTemplate it fails and does not create the slave.
Below is the log:
```
WARNING: Error in provisioning; agent=KubernetesSlave name: jenkins-slave-rr5sj-q58nt, template=PodTemplate{inheritFrom='base', name='jenkins-slave-rr5sj', namespace='rbehera-1-jenkins', label='nodejs_health_check_redhat_1', serviceAccount='jenkins', volumes=[SecretVolume [mountPath=/home/jenkins/.gnupg-ro, secretName=jenkins-release-gpg], SecretVolume [mountPath=/home/jenkins/.apitoken, secretName=jenkins-hub-api-token], SecretVolume [mountPath=/root/.ssh-ro, secretName=jenkins-ssh-config], SecretVolume [mountPath=/root/.ssh-git-ro, secretName=jenkins-git-ssh]], containers=[ContainerTemplate{name='node', image='openshift/jenkins-agent-nodejs-8-centos7', workingDir='/home/jenkins/', command='/bin/bash -c', args='cat', ttyEnabled=true, resourceLimitMemory='640Mi'}, ContainerTemplate{name='jnlp', image='fabric8/jenkins-slave-base-centos7:v2132422', workingDir='/home/jenkins/', args='${computer.jnlpmac} ${computer.name}', resourceLimitMemory='256Mi'}]}
java.lang.IllegalStateException: Agent is not connected after 100 attempts, status: Running
  at org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher.launch(KubernetesLauncher.java:190)
  at hudson.slaves.SlaveComputer$1.call(SlaveComputer.java:294)
  at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
  at jenkins.security.ImpersonatingExecutorService$2.call(ImpersonatingExecutorService.java:71)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
```